designs/v1-roadmap.md

```diff
@@ -0,0 +1,208 @@
+3. Planned Deprecations: The feature cleans-up deprecations that were previously planned the project
+
+## Roadmap
+
+1. [v1 APIs](#v1-apis)
+2. [Stabilize Observability (metrics, status, eventing)](#stabilize-observability--metrics-status-eventing-)
+3. [Stabilize Karpenter’s Tainting Logic](#stabilize-karpenters-tainting-logic)
+4. [Wait for Instance Termination on NodeClaim/Node Deletion](#wait-for-instance-termination-on-nodeclaimnode-deletion)
+5. [Drift Hash Breaking Change Handling](#drift-hash-breaking-change-handling)
+6. [Removing Ubuntu AMIFamily](#removing-ubuntu-amifamily-aws-cloudprovider)
+7. [Introduce ConsolidateAfter for Consolidation Controls](#introduce-consolidateafter-for-consolidation-controls)
+8. [Define SemVer Versioning Policy for kubernetes-sigs/karpenter Library](#define-semver-versioning-policy-for-kubernetes-sigskarpenter-library)
+9. [NodeClaim Conceptual Documentation](#nodeclaim-conceptual-documentation)
+10. [Drop Knative References from the Code](#drop-knative-references-from-the-code)
+11. [Migrate Knative Webhook away from Karpenter](#migrate-knative-webhook-away-from-karpenter)
+12. [Karpenter Global Logging Configuration Changes](#karpenter-global-logging-configuration-changes-aws-cloudprovider)
+13. [Promoting Drift Feature to Stable](#promoting-drift-feature-to-stable)
+14. [Removing Implicit ENI Public IP Configuration](#removing-implicit-eni-public-ip-configuration-aws-cloudprovider)
+
+### v1 APIs
+
+**Issue Ref(s):** https://github.com/kubernetes-sigs/karpenter/issues/758, https://github.com/aws/karpenter-provider-aws/issues/5006
+
+**Category:** Breaking, Stability
+
+For Karpenter to be considered v1, the CustomResources that are shipped with an installation of the project also need to be stable at v1. Changes to Karpenter’s API (including labels, annotations, and tags) in v1 are detailed in [Karpenter v1 API](./v1-api.md). The migration path for these changes will ensure that customers will not have to roll their nodes or manually convert their resources as they did at v1beta1. Instead, we will leverage Kubernetes [conversion webhooks](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#webhook-conversion) to automatically convert their resources to the new schema format in code. The API groups and Kind naming will remain unchanged.
+
+### Stabilize Observability (metrics, status, eventing)
+
+**Issue Ref:** https://github.com/kubernetes-sigs/karpenter/issues/1051
+
+**Category:** Breaking, Stability
+
+Karpenter needs to stabalize a set of metrics, status conditions, and events that can be relied-upon for monitoring. The design for these metrics, status conditions and events will be added in a separate RFC.
+
+### Stabilize Karpenter’s Tainting Logic
+
+**Issue Ref:** https://github.com/kubernetes-sigs/karpenter/issues/624, https://github.com/kubernetes-sigs/karpenter/issues/1049
+
+**Category:** Breaking
```

Comment:
We shouldn't align on `node-lifecycle.kubernetes.io/do-not-disrupt` until it's registered with Kubernetes (see https://kubernetes.io/docs/reference/labels-annotations-taints/ for the list  - you just need a PR against that page to add a registered thing).

So, an easy prerequisite to meet, but still something to remember to do.

Reply:
> do we plan on [aligning](https://docs.google.com/document/d/1rHhltfLV5V1kcnKr_mKRKDC4ZFPYGP4Tde2Zy-LE72w/edit) with cas on some of the taints and annotations for v1 as well? Or is this work still in discussion phase

There is an [open issue on this upstream.](https://github.com/kubernetes/autoscaler/issues/6648) We've been actively having some conversations around what makes sense here, but I imagine that, yes, this will be part of the taint redesign so that we don't break everyone twice.

Since the discussion was still ongoing and nothing was set in stone yet, I didn't want to specifically scope it into this item; though, I can add a little bit to make mention to it.